File: saransh_kunal-LINUX/Assignment/dotnet/Library/Loan/LoanLib/HomeLoan.cs
namespace Loaning;

sealed class HomeLoan : Loan
{
    public override float Rate(double amount)
    {
        return principle <= 2000000 ? 0.10f:0.12f ;
    }

    public override double EMI()
    {
        return principle*(1+ Rate(principle) * period/100) / (12*period);
        //emi = p * (1 + r * n / 100) / (12 * n)
    }

    public double GetDiscount()
    {
        return EMI() > 1000000 ? 0.95 * principle : 0;
    }

    public double GetTotalDiscount()
    {
       return 0.0; 
    }
};

File: saransh_kunal-LINUX/Assignment/java/Assignment_3/loan/HomeLoan.java
package loan;

public final class HomeLoan extends Loan implements Discountable {
    public HomeLoan(double p, double n){
        super(p, n);
    }

    @Override
    public double GetDiscount() {
        return 0.05;
    }

    @Override
    public double GetInterestRate() {
        return principle <= 2000000 ? 10 : 11;
    }
}
